distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Postgres/Postgres_Dialect.enso

```diff
@@ -480,6 +482,53 @@ decimal_div = Base_Generator.lift_binary_op "DECIMAL_DIV" x-> y->
 decimal_mod = Base_Generator.lift_binary_op "DECIMAL_MOD" x-> y->
     x ++ " - FLOOR(CAST(" ++ x ++ " AS decimal) / CAST(" ++ y ++ " AS decimal)) * " ++ y
 
+replace_text_cs_all = Base_Generator.lift_ternary_op "REPLACE_TEXT_CS_ALL" t-> p-> r->
+    Builder.code "REPLACE(" ++ t ++ ", " ++ p ++ ", " ++ r ++ ")"
+
+replace_text_cs_only_first = Base_Generator.lift_ternary_op "REPLACE_TEXT_CS_ONLY_FIRST" t-> p-> r->
+    escaped_pattern = escape_regex_param p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ escaped_pattern ++ ", " ++ r ++ ", 1, 1)"
+
+replace_text_ci_all = Base_Generator.lift_ternary_op "REPLACE_TEXT_CI_ALL" t-> p-> r->
+    escaped_pattern = escape_regex_param p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ escaped_pattern ++ ", " ++ r ++ ", 1, 0, 'i')"
+
+replace_text_ci_only_first = Base_Generator.lift_ternary_op "REPLACE_TEXT_CI_ONLY_FIRST" t-> p-> r->
+    escaped_pattern = escape_regex_param p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ escaped_pattern ++ ", " ++ r ++ ", 1, 1, 'i')"
+
+replace_regex_cs_all = Base_Generator.lift_ternary_op "REPLACE_REGEX_CS_ALL" t-> p-> r->
+    pattern_string = transform_literal_param (_.pattern_string) p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ pattern_string ++ ", " ++ r ++ ", 1, 0)"
+
+replace_regex_cs_only_first = Base_Generator.lift_ternary_op "REPLACE_REGEX_CS_ONLY_FIRST" t-> p-> r->
+    pattern_string = transform_literal_param (_.pattern_string) p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ pattern_string ++ ", " ++ r ++ ", 1, 1)"
+
+replace_regex_ci_all = Base_Generator.lift_ternary_op "REPLACE_REGEX_CI_ALL" t-> p-> r->
+    pattern_string = transform_literal_param (_.pattern_string) p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ pattern_string ++ ", " ++ r ++ ", 1, 0, 'i')"
+
+replace_regex_ci_only_first = Base_Generator.lift_ternary_op "REPLACE_REGEX_CI_ONLY_FIRST" t-> p-> r->
+    pattern_string = transform_literal_param (_.pattern_string) p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ pattern_string ++ ", " ++ r ++ ", 1, 1, 'i')"
+
+replace_column_cs_all = Base_Generator.lift_ternary_op "REPLACE_COLUMN_CS_ALL" t-> p-> r->
+    Builder.code "REPLACE(" ++ t ++ ", " ++ p ++ ", " ++ r ++ ")"
+
+## PRIVATE
+   Escape a literal regex param
+escape_regex_param : Builder -> Builder
+escape_regex_param regex_param = transform_literal_param Regex.escape regex_param
+
+## PRIVATE
+   Transform the interpolation value within the Builder
+transform_literal_param : (Any -> Any) -> Builder -> Builder
+transform_literal_param transformer param =
+    x = case param of Builder.Value (Vector_Builder.Leaf x) -> case x.at 0 of SQL_Fragment.Interpolation y -> y
+    transformed_x = transformer x
+    Builder.Value (Vector_Builder.Leaf [SQL_Fragment.Interpolation transformed_x])
```

Comment:
I don't like this hack. It assumes a lot about the structure of generated IR.

If we really need it - OK.

But I don't think we do.

I think you are passing the `pattern_string` already, so there should be no need to do the transform, right?

Reply:
We do really need this, since we are choosing whether to modify the parameter after it has been embedded in the DSL wrapper, I believe.

Reply:
Actually, we can still do it without this - and IMHO it would be better, but I don't want to be stubborn, I think both approaches have their + and - so I leave the decision to you.

If you want to keep this function, I'd rename it to `transform_interpolated_param`. `SQL_Expression.Literal` is a _different_ thing so currently this name is misleading.

------

How we can deal without this `transform_`?

You can keep the parameter to be used by default but also add the 'raw' pattern string as part of metadata. In the branch where you need to transform it (escape the regex), you can get the raw text from the metadata. To insert it back into the code without risk of SQL injection, you can use `Builder.interpolation` to insert an interpolated `?` parameter instead of raw code.

Reply:
Good idea, done.